reframe/schemas/config.json

```diff
@@ -362,10 +361,15 @@
                         "type": "array",
                         "items": {"type": "string"}
                     },
-                    "extras": {"type": "object"},
+                    "extras": {
+                        "type": "object",
+                        "propertyNames": {
+                            "pattern": "^[a-zA-Z_][a-zA-Z0-9_]*$"
+                        }
+                    },
                     "features": {
                         "type": "array",
-                        "items": {"type": "string"}
+                        "items": {"$ref": "#/defs/alphanum_ext_string"}
```

Comment:
I think we want the same pattern as the extras, right? Do we want to allow `-` in features?

Reply:
That's probably an omission.

Reply:
> I think we want the same pattern as the extras, right?

No, extras' value can be non-strings. Only the keys must follow a certain pattern.

 > Do we want to allow `-` in features?

Correct point, we can't. I will change it to simple alphanum.

Reply:
Yes, I didn't write it properly but this is what I meant. To set the same pattern to extras' keys and feautures. I agree that extras' value should remain object.

Reply:
I explicitly restricted it a bit more, so as to a valid Python identifier in case we need to create easily an object out of it in the future.